test/unit/metrics/test_metricframe_bootstrap_values.py

```diff
@@ -0,0 +1,180 @@
+
+# Absolute number of group 'A' which are correct
+n_A_correct = int(p_A * n_A)
+
+# And for group 'B'
+n_B_correct = int(n_samples * p) - n_A_correct
+p_B = n_B_correct / n_B
+
+# Absolute numbers which are incorrect
+n_A_incorrect = n_A - n_A_correct
+n_B_incorrect = n_B - n_B_correct
+
+
+@pytest.fixture(scope="session")
+def mf():
+    y_true = np.ones(n_samples)
+    y_pred = np.concatenate(
+        (np.ones(int(n_samples * p)), np.zeros(int(n_samples * (1 - p))))
+    )
+    # Check for rounding errors
+    assert len(y_pred) == n_samples
+    assert n_A + n_B == n_samples
+
+    # Make sure the overall metric is as expected
+    assert accuracy_score(y_true, y_pred) == p
+
+    # Sanity check
+    assert n_A_correct < n_samples * p
+
+    s_f = np.concatenate(
+        (
+            np.full(n_A_correct, "A"),
+            np.full(n_B_correct, "B"),
+            np.full(n_A_incorrect, "A"),
+            np.full(n_B_incorrect, "B"),
+        )
+    )
+
+    # Make sure we have avoided rounding errors
+    assert len(s_f) == n_samples
+
+    # Create the metric frame
+    # The ci argument is set to have the
+    # nominal value, as well as the one and two
+    # sigma values
+    mf = MetricFrame(
+        metrics=accuracy_score,
+        y_true=y_true,
+        y_pred=y_pred,
+        sensitive_features=s_f,
+        n_boot=10000,
+        ci=(
+            0.025,
+            0.1587,
+            0.5,
+            0.8413,
+            0.975,
+        ),  # -2sigma, -sigma, nominal, sigma, 2sigma
+    )
+
+    return mf
+
+
+class TestBootstrapValues:
+    def test_overall(self, mf):
+        assert mf.overall == p
+
+    def test_by_group(self, mf):
+        assert mf.by_group["A"] == p_A
+        assert mf.by_group["B"] == p_B
+
+    def test_overall_ci(self, mf):
+        mean = p * n_samples
+        sigma = math.sqrt(n_samples * p * (1 - p))
+
+        # Following corresponds to the ci argument to MetricFrame
+        sigma_vals = [-2, -1, 0, 1, 2]
+
+        for i in range(len(sigma_vals)):
+            value = mean + (sigma_vals[i] * sigma)
+            value_frac = value / n_samples
+
+            assert mf.overall_ci[i][1] == pytest.approx(
+                value_frac, rel=1 / math.sqrt(n_samples)
+            )
+
+    def test_by_group_ci(self, mf):
+        mean_A = p_A * n_A
+        sigma_A = math.sqrt(n_A * p_A * (1 - p_A))
+
+        mean_B = p_B * n_B
+        sigma_B = math.sqrt(n_B * p_B * (1 - p_B))
+
+        # Following corresponds to the ci argument to MetricFrame
+        sigma_vals = [-2, -1, 0, 1, 2]
+
+        for i in range(len(sigma_vals)):
+            v_A = mean_A + (sigma_vals[i] * sigma_A)
+            vf_A = v_A / n_A
+
+            v_B = mean_B + (sigma_vals[i] * sigma_B)
+            vf_B = v_B / n_B
+
+            # Slightly wider tolerance here.... not sure if this should be required
+            assert mf.by_group_ci[i][1]["A"] == pytest.approx(
+                vf_A, rel=2 / math.sqrt(n_A)
```

Comment:
@MiroDudik and @Harsha-Nori, your feedback here would be much appreciated. This is a wider tolerance than I was expecting to need, so I'm worried that I'm doing something wrong in the calculation.

Reply:
Hmm.... this has been lost from the 'Files' tab unfortunately :-/


I think I get what you're saying here @MiroDudik , but I'd like to be certain.... Basically, what I was aiming to do here was verify that sensible values were being produced, because nothing in the existing tests did that. What you're saying @MiroDudik  is that the way I've gone about this is also testing whether or not bootstrapping works statistically (because I'm relying on certain properties of binomial and normal distributions, and then hammering down uncertainties with a large n_boot)? So a better approach would be to create a 'known' set of bootstrap samples, and then perform the quantile calculations on those?